make/common.mk

```diff
@@ -154,6 +157,16 @@ clean:
 
 # Make flash
 flash:
+ifeq ($(FLASHING_TOOL), st-flash)
 	st-flash write $(BIN_FOLDER)/$(BIN_FILE_NAME) $(FLASH)
+else ifeq ($(FLASHING_TOOL), stm32flash)
+ifdef FLASHING_SERIAL_PORT
```

Comment:
Can you move these checks to the top of the makefile? So you would end up with something like this:
```
ifeq ($(FLASHING_TOOL), stm32flash)
  ifndef FLASHING_SERIAL_PORT
    $(error Please set the required FLASHING_SERIAL_PORT variable in your makefile)
  endif
endif
```

This would make the recipe for `flash` way easier to read.

Reply:
I'd like to have configuration error checking at the top of the makefile, just to keep things easier to manage and read.